### DISCUSSION TITLE
NVIDIA and all the other big models are installed with `pip`

### DISCUSSION THREAD

**Discussion by SamuelMarks** [open-webui/open-webui#4286] [General]
**Is your feature request related to a problem? Please describe.**
My build is huge! - 6.8G from a `pip install open-webui`  + your `open-webui/backend/requirements.txt`.

**Describe the solution you'd like**
`pip install open-webui` could be:
- `pip install open-webui[nothing]` for just using API keys (nothing local), or
- `pip install open-webui[rocm]` or 
- `pip install open-webui[cuda]`.

**Describe alternatives you've considered**
Crying / buying new harddrive / manually removing the big files I don't need

**Additional context**
N/A

**Comment by EncryptedCurse** [comment]
Hopefully we can get some movement on this...I shouldn't need to download 1.5GB+ of NVIDIA-related packages on a system without a GPU.
